certbot/main.py

```diff
@@ -405,6 +406,65 @@ def _init_le_client(config, authenticator, installer):
 
     return client.Client(config, acc, authenticator, installer, acme=acme)
 
+def _delete_if_appropriate(config):
+    """Does the user want to delete their now-revoked certs?"""
+
+    display = zope.component.getUtility(interfaces.IDisplay)
+
+    if not (config.certname or config.cert_path):
+        raise errors.Error('At least one of --cert-path or --cert-name must be specified.')
+
+    if config.certname and config.cert_path:
+        # first, check if certname and cert_path imply the same certs
+        cert_path_implied_cert_name = cert_manager.cert_path_to_lineage(config)
```

Comment:
yup, exactly. because `cert_path_to_lineage` searches through all certs, whereas `lineage_for_certname` is constant time. and it turns out we do have users with hundreds of thousands of certs in a single certbot installation.